.travis.yml

```diff
@@ -2,8 +2,45 @@ language: c
 dist: trusty
 sudo: required
 before_install:
-  # the following packages are required by some of the packages
-  # installting them via "opam depext" is possible, too
-  - sudo apt-get install hwdata libpci-dev libxen-dev
-script:
-  - sh travis.sh
+    - sudo add-apt-repository -y "deb http://archive.ubuntu.com/ubuntu/ trusty main universe"
+    - sudo apt-get update -qq
+    - sudo apt-get install libgcrypt20 liblzma5 libselinux1
+    - wget "http://archive.ubuntu.com/ubuntu/ubuntu/pool/main/s/systemd/libsystemd0_219-7ubuntu3_amd64.deb"
+    - wget "http://archive.ubuntu.com/ubuntu/ubuntu/pool/main/s/systemd/libsystemd-dev_219-7ubuntu3_amd64.deb"
+    - sudo dpkg -i libsystemd0_219-7ubuntu3_amd64.deb libsystemd-dev_219-7ubuntu3_amd64.deb
+script: bash -ex travis.sh
+env:
+  global:
+    - OCAML_VERSION=4.02
+  matrix:
+    - BASE_REMOTE=$TRAVIS_BUILD_DIR
+    - BASE_REMOTE=$TRAVIS_BUILD_DIR OPAM_LINT=1
+    - BASE_REMOTE=$TRAVIS_BUILD_DIR COMPILE_ALL=1
+    - EXTRA_REMOTES=$TRAVIS_BUILD_DIR COMPILE_ALL=1
+    - OCAML_VERSION=4.04 EXTRA_REMOTES=$TRAVIS_BUILD_DIR COMPILE_ALL=1
+matrix:
+  fast_finish: true
+  allow_failures:
+    - env: BASE_REMOTE=$TRAVIS_BUILD_DIR OPAM_LINT=1
+    - env: BASE_REMOTE=$TRAVIS_BUILD_DIR COMPILE_ALL=1
+    - env: EXTRA_REMOTES=$TRAVIS_BUILD_DIR COMPILE_ALL=1
+    - env: OCAML_VERSION=4.04 EXTRA_REMOTES=$TRAVIS_BUILD_DIR COMPILE_ALL=1
+addons:
+  apt:
+    packages:
```

Comment:
I had issues using depext because of conflicts with `libsystemd-dev`, we could try to revert to it.